File: entsoe-client/src/main/java/org/example/model/common/CurveType.java
package org.example.model.common;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Arrays;

@Getter
@AllArgsConstructor
public enum CurveType {
    SEQUENTIAL_FIXED_SIZE_BLOCK("A01", "Sequential fixed size block"),
    POINT_TO_POINT("A02", "Point to point"),
    VARIABLE_SIZED_BLOCK("A03", "Variable sized block");

    private final String code;
    private final String description;

    public static CurveType fromCode(String code) {
        return Arrays.stream(values())
                .filter(type -> type.code.equals(code))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Unknown CurveType code: " + code));
    }
}

File: entsoe_client/src/entsoe_client/model/common/curve_type.py
from enum import Enum
from typing import Self

from entsoe_client.exceptions.unknown_curve_type_error import (
    UnknownCurveTypeError,
)


class CurveType(Enum):
    SEQUENTIAL_FIXED_SIZE_BLOCK = ("A01", "Sequential fixed size block")
    POINT_TO_POINT = ("A02", "Point to point")
    VARIABLE_SIZED_BLOCK = ("A03", "Variable sized block")

    def __init__(self, code: str, description: str) -> None:
        self.code = code
        self.description = description

    @classmethod
    def from_code(cls, code: str) -> Self:
        for member in cls:
            if member.code == code:
                return member
        raise UnknownCurveTypeError(code)
